app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/Matchers.ts

```diff
@@ -0,0 +1,170 @@
+import { Platform } from 'react-native';
+import { color, defaultTestErrorLog } from './logMessageUtils';
+import { ComparisonMode, TestCase, TrackerCallCount } from './types';
+
+type TestValue =
+  | TrackerCallCount
+  | string
+  | Array<unknown>
+  | number
+  | bigint
+  | Record<string, unknown>;
+
+export class Matchers {
+  constructor(private currentValue: TestValue, private testCase: TestCase) {}
+
+  private assertValueIsCallTracker(
+    value: any
+  ): asserts value is TrackerCallCount {
+    if (!('name' in value && 'JS' in value && 'UI' in value)) {
+      throw Error('Invalid value');
+    }
+  }
+
+  public toBe(
+    expectedValue: TestValue,
+    comparisonMode = ComparisonMode.DISTANCE
+  ) {
+    const CONDITIONS: { [Key: string]: (e: any, v: any) => Boolean } = {
+      [ComparisonMode.STRING]: (expected, value) => {
+        return value !== expected;
+      },
+      [ComparisonMode.NUMBER]: (e, v) => {
+        return isNaN(Number(v)) ? !isNaN(e) : Number(v) !== Number(e);
+      },
+      [ComparisonMode.COLOR]: (e, v) => {
+        const colorRegex = new RegExp('^#?([a-f0-9]{6}|[a-f0-9]{3})$');
+        if (!colorRegex.test(expectedValue as string)) {
+          throw Error(
+            `Invalid color format "${e}", please use lowercase hex color (like #123abc) `
+          );
+        }
+        return typeof v !== 'string' || v !== e;
+      },
+      [ComparisonMode.DISTANCE]: () => {
+        const valueAsNumber = Number(this.currentValue);
+        if (Platform.OS === 'ios') {
+          return (
+            isNaN(valueAsNumber) || valueAsNumber !== Number(expectedValue)
```

Comment:
Can we really not include a margin of error here?

Reply:
We need a "margin of error" since on Android we get all sizes in pixels. So the actual amount of pixels displaying certain element is always an integer, and then we convert it into dpi.

$$ px=dp \cdot(dpi/160)$$

$$ dp=px \cdot (160/dpi)$$

Since value of $px$ gets rounded, we have an error of $160/dpi$, and the baseline density is exactly 160, so I've decided to set error of $\pm 1$.

@piaskowyk Actually we should use https://reactnative.dev/docs/pixelratio#get here to adjust this setting for devices with non-default density

Reply:
Yes, but here you don't seem to take this margin of error into account, no?